src/ResourceManager/Compute/Commands.Compute/Strategies/ComputeRp/ImageEx.cs

```diff
@@ -81,10 +84,44 @@ private static int[] CreatePorts(this OperatingSystemTypes osType)
                     location, image.Publisher, image.Offer, image.Sku, image.Version);
                 return new ImageAndOsType(imageModel.OsDiskImage.OperatingSystem, image);
             }
+            else if (imageName.Contains("/"))
+            {
+                var resourceId = ResourceId.TryParse(imageName);
+                if (resourceId == null
+                    || resourceId.ResourceType.Namespace != ComputeStrategy.Namespace
+                    || resourceId.ResourceType.Provider != "images")
+                {
+                    throw new ArgumentException(string.Format(Resources.ComputeInvalidImageName, imageName));
+                }
+
+                if (compute.SubscriptionId != resourceId.SubscriptionId)
+                {
+                    throw new ArgumentException(Resources.ComputeMismatchSubscription);
+                }
+
+                var localImage = await compute.Images.GetAsync(
+                    resourceGroupName: resourceId.ResourceGroupName,
+                    imageName: resourceId.Name);
+
+                return new ImageAndOsType(
+                    localImage.StorageProfile.OsDisk.OsType,
+                    new ImageReference { Id = localImage.Id });
+            }
             else
             {
-                // get image
-                return Images
+                try
+                {
+                    var localImage = await compute.Images.GetAsync(resourceGroupName, imageName);
```

Comment:
@sergey-shandar does this address Mark's comments?

_We shouldn't duplicate this code -it feels like this should be in the same if case as if the image contains a '/'. I would suggest first checking if the name contains a ':', then a single case for '/' or no '/'._

From the comment, it seems like the above `else if` should just be an `else`.

Reply:
The implementations are quite different and we don't want to invalid image ids to pass as image names.